app/models/post_action_type.rb

```diff
@@ -11,51 +11,59 @@ def expire_cache
     ApplicationSerializer.expire_cache_fragment!(/\Apost_action_flag_types_/)
   end
 
+  DiscourseEvent.on(:reload_post_action_types) { self.reload_types }
+
   class << self
     attr_reader :flag_settings
 
-    def replace_flag_settings(settings)
-      if settings
-        @flag_settings = settings
-      else
-        initialize_flag_settings
-      end
-      @types = nil
+    def initialize_flag_settings
+      @flag_settings = FlagSettings.new
     end
 
-    def ordered
-      order("position asc")
+    def replace_flag_settings(settings)
```

Comment:
Might just want to make a comment here about how this is deprecated, so people don't use it directly and are aware of the coming plugin deprecation.